libraries/AR_Motors/AP_MotorsUGV.h

```diff
@@ -221,9 +223,25 @@ class AP_MotorsUGV {
     float   _throttle_factor[AP_MOTORS_NUM_MOTORS_MAX];
     float   _steering_factor[AP_MOTORS_NUM_MOTORS_MAX];
     float   _lateral_factor[AP_MOTORS_NUM_MOTORS_MAX];
-    uint8_t   _motors_num;
+    uint8_t    _motors_num;
+    uint32_t    _last_set_throttle_call;   // used by esc_backward_unlock logic
+    float    _esc_backward_unlock_simulated_signal;  // signal sent to esc to simulate backward unlock
+    bool    _esc_backward_unlock_active;  // esc unlock process in active     
+    UnlockState    _esc_backward_unlock_state;  // state machine status for backward unlock logic.
 
     static AP_MotorsUGV *_singleton;
+
+private:
+    // state machine status for backward unlock logic.
+    enum UnlockState { 
+        UNLOCK_WAITING = 0,
+        UNLOCK_STEP_DEC_TO_MAX_NEG = 1,
+        UNLOCK_STEP_STAY_MAX = 2,
+        UNLOCK_STEP_INC_TO_ZERO = 3,
+        UNLOCK_STEP_ZEROS_2 = 4,
+        UNLOCK_FINISHED = 5
+    } _esc_backward_unlock_state ; 
```

Comment:
```suggestion
    } ; 
```

Reply:
You wilt have to put it above the struct I think in this case, but I think you can move the struct to private.

Reply:
This variable is unused.

Reply:
@HefnySco ^